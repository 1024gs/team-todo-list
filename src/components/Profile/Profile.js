import React from 'react';
import './Profile.scss';

const Profile = ({user}) => {
  return (
    <div className="Profile">
      <div className="Profile-avatar">
        <i className="material-icons">account_circle</i>
      </div>
      <div className="Profile-username">{user.name}</div>
    </div>
  );
};

export default Profile;
