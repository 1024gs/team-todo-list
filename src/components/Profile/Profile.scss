.Profile {
  display: flex;
  height: 70px;
}

.Profile .Profile-avatar,
.Profile .Profile-username {
  line-height: 70px;
}

.Profile-avatar {
  margin-right: 10px;
}

.Profile-avatar .material-icons {
  line-height: inherit;
  font-size: 50px;
  color: #ccc;
}

.Profile-username {
  flex-grow: 1;
}
