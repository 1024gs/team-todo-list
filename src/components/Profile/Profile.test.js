import React from 'react';
import Profile from './Profile';
import {shallow} from 'enzyme';

const setup = (propOverrides) => {
  const props = Object.assign({user: {name: 'Guglielmo Marconi'}}, propOverrides);
  const wrapper = shallow(<Profile {...props} />);

  return {
    props,
    wrapper
  };
};

describe('Profile', () => {
  it('renders', () => {
    const {wrapper} = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
