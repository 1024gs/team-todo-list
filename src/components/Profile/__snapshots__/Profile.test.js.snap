// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders 1`] = `
<div
  className="Profile"
>
  <div
    className="Profile-avatar"
  >
    <i
      className="material-icons"
    >
      account_circle
    </i>
  </div>
  <div
    className="Profile-username"
  >
    Guglielmo Marconi
  </div>
</div>
`;
